pkg/apis/batch/validation/validation.go

```diff
@@ -430,15 +434,24 @@ func validateJobStatus(status *batch.JobStatus, fldPath *field.Path) field.Error
 
 // ValidateJobUpdate validates an update to a Job and returns an ErrorList with any errors.
 func ValidateJobUpdate(job, oldJob *batch.Job, opts JobValidationOptions) field.ErrorList {
-	allErrs := apivalidation.ValidateObjectMetaUpdate(&job.ObjectMeta, &oldJob.ObjectMeta, field.NewPath("metadata"))
+	allErrs := validateJobMetaUpdate(&job.ObjectMeta, &oldJob.ObjectMeta)
 	allErrs = append(allErrs, ValidateJobSpecUpdate(job.Spec, oldJob.Spec, field.NewPath("spec"), opts)...)
 	return allErrs
 }
 
 // ValidateJobUpdateStatus validates an update to the status of a Job and returns an ErrorList with any errors.
 func ValidateJobUpdateStatus(job, oldJob *batch.Job) field.ErrorList {
-	allErrs := apivalidation.ValidateObjectMetaUpdate(&job.ObjectMeta, &oldJob.ObjectMeta, field.NewPath("metadata"))
-	allErrs = append(allErrs, ValidateJobStatusUpdate(job.Status, oldJob.Status)...)
+	allErrs := validateJobMetaUpdate(&job.ObjectMeta, &oldJob.ObjectMeta)
+	allErrs = append(allErrs, ValidateJobStatusUpdate(job, oldJob)...)
+	return allErrs
+}
+
+func validateJobMetaUpdate(newMeta, oldMeta *metav1.ObjectMeta) field.ErrorList {
+	fldPath := field.NewPath("metadata")
+	allErrs := apivalidation.ValidateObjectMetaUpdate(newMeta, oldMeta, fldPath)
+	if utilfeature.DefaultFeatureGate.Enabled(features.JobManagedByLabel) {
```

Comment:
> Because this isn't a new field and labels are being used instead of field selectors (why is this actually?) and because the strategy isn't preventing sets of the field when the featuregate is off, this doesn't actually prevent mutation of the field in question.

agreed, don't check feature gates in validation

Reply:
> agreed, don't check feature gates in validation

Done, now I'm not checking the feature gate in this validation for immutability of the label.

> field selectors (why is this actually?) 

The reasons for not using label / field selectors are discussed in KEP:https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/4368-support-managed-by-label-for-batch-jobs#using-label-selectors. Essentially: 
- they don't make much of a difference from the memory point of view, because k8s is using shared informers
- there is no support for OR, and we wanted to skip if no label, or `job-controller.k8s.io`. For field we would have a similar dillema I guess
- they don't let us to log "job skipped" which might be handy for debugging purposes.